build/generate_crds.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright 2023 The Kanister Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+export GO111MODULE=on
+
+## Location to install dependencies to
+LOCALBIN=$(pwd)/bin
+## Tool Binaries
+CONTROLLER_GEN=${LOCALBIN}/controller-gen
+## Tool Versions
+CONTROLLER_TOOLS_VERSION=${1}
+
+test -s ${CONTROLLER_GEN} || GOBIN=${LOCALBIN} go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
+${CONTROLLER_GEN} crd webhook paths="github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1" output:crd:artifacts:config=pkg/customresource
```

Comment:
- How are we generating Kanister CRDs currently?
- What changes would it need to generate Kanister CRDs with controller-gen?

I believe we should follow a common pattern for all the CRDs?

Reply:
@PrasadG193 This generates all the CRDs under package `github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1`. I dont see any target in the makfile which generates kanister CRDs apart from the one I have written, We have `make codegen`, but that  just generates the code and not the CRDs. Hence I think this would be the right target to generate all the CRDs

Reply:
Let me move the target to the main Makefile rather than kubebuilder`s makefile

Reply:
Done. let me know if we should merge this PR now?